File: frontend/react-paylateapp/src/models/UserOrder.ts
import OrderItem from "./OrderItem";

class UserOrder {
    id: number;
    orderDate: string;
    status: string;
    items: OrderItem[];
    userEmail: string;

    constructor(id: number, orderDate: string, status: string, items: OrderItem[],
        userEmail: string) {
        this.id = id;
        this.orderDate = orderDate;
        this.status = status;
        this.items = items;
        this.userEmail = userEmail;
    }

    getTotalAmount(): number {
        return this.items.reduce((total, item) => total + item.price * item.quantity, 0);
    }
}

export default UserOrder;

File: backend/order-service/src/main/java/org/payLate/entity/UserOrder.java
package org.payLate.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Data
public class UserOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "user_email")
    private String userEmail;

    @CreationTimestamp
    @Column(name = "user_order_date", updatable = false)
    private Date orderDate;

    @Column(name = "status")
    private String status;

    @OneToMany(mappedBy = "userOrder", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<OrderItem> items = new ArrayList<>();
}